### ISSUE TITLE
TODB should cascade delete interfaces and IP addresses when deleting a server via SQL

### ISSUE THREAD

**Issue by rawlinp** [apache/trafficcontrol#5364]
## I'm submitting a ...
-  bug report

## Traffic Control components affected ...
-  Traffic Ops

## Current behavior:
When trying to delete a server via `DELETE FROM server WHERE ...`, you are restricted due to the `ON DELETE RESTRICT` FK clauses on the `interface` and `ip_address` tables. Additionally, deleting server interfaces is restricted via the `ON DELETE RESTRICT` FK clause on the `ip_address` table.

## Expected behavior:
If you delete a server via `DELETE FROM server WHERE ...`, it should cascade delete all the server's associated interfaces and IP addresses. If you delete an interface, it should cascade delete all the interface's associated IP addresses.

Basically these FK clauses should be `ON DELETE CASCADE` instead of `ON DELETE RESTRICT`.

## Minimal reproduction of the problem with instructions:
For a server that has at least one interface, attempt to delete it via the SQL query: `DELETE FROM server WHERE id = ...`. For an interface with at least one IP address, attempt to delete it via the SQL query: `DELETE FROM interface WHERE server = ... AND name = ...`.
